.circleci/config.yml

```diff
@@ -15,25 +15,33 @@ common: &common
     - run:
         name: build geth if missing
         command: |
-          mkdir -p $HOME/.ethash
-          pip install --user py-geth>=1.10.1
-          export GOROOT=/usr/local/go
-          export GETH_VERSION=v1.7.2
-          export GETH_BINARY="$HOME/.py-geth/geth-$GETH_VERSION/bin/geth"
-          if [ ! -e "$GETH_BINARY" ]; then
-            curl -O https://storage.googleapis.com/golang/go1.7.4.linux-amd64.tar.gz
-            tar xvf go1.7.4.linux-amd64.tar.gz
-            sudo chown -R root:root ./go
-            sudo mv go /usr/local
-            sudo ln -s /usr/local/go/bin/go /usr/local/bin/go
-            sudo apt-get update;
-            sudo apt-get install -y build-essential;
-            python -m geth.install $GETH_VERSION;
-            sudo ln -s /home/circleci/.py-geth/geth-v1.7.2/bin/geth /usr/local/bin/geth
+          if [ -n "$GETH_VERSION" ]; then
```

Comment:
I seem to remember there being a problem with doing this, like it messed up the cache. Maybe it would work if you add the GETH_VERSION environment variable to the cache key. Note that because the cache was already built successfully, changes here are dangerous: we might break things and not realize it.